server/src/lib/basket.ts

```diff
@@ -11,80 +11,154 @@ export const API_URL =
               AND DATEDIFF(NOW(), last_activity_at) <= 1
           )
         ) AS two_day_streak
-      FROM user_clients
-      LEFT JOIN custom_goals goals ON user_clients.client_id = goals.client_id
+      FROM user_client_newsletter_prefs AS newsletter
+      LEFT JOIN custom_goals goals ON newsletter.client_id = goals.client_id
                                       AND goals.locale_id = 1
       LEFT JOIN custom_goals current_goal ON (
-        user_clients.client_id = current_goal.client_id AND
+        newsletter.client_id = current_goal.client_id AND
         current_goal.created_at >= goals.created_at
       )
       LEFT JOIN awards ON current_goal.id = awards.custom_goal_id
-      WHERE user_clients.client_id = ? AND has_login
-      GROUP BY user_clients.client_id
+      WHERE newsletter.client_id = ?
+      GROUP BY newsletter.client_id
     `,
-    [client_id]
+    [currentUserStats.client_id]
   );
-  if (
-    !row ||
-    !row.basket_token ||
-    (!row.first_contribution_date && !row.goal_created_at)
-  ) {
-    return;
-  }
-  const data = {
-    email: row.email,
 
-    first_contribution_date: toISO(row.first_contribution_date),
+  db.query(
+    `
+      UPDATE user_client_newsletter_prefs
+      SET 
+        first_contrib = ?,
+        goal_created = ?, 
+        goal_reached = ?,
+        two_day_streak = ?, 
+        last_active = ?
+      WHERE client_id = ?`,
+    [
+      computed.first_contribution_date,
+      computed.goal_created_at,
+      computed.goal_reached_at,
+      Boolean(computed.two_day_streak),
+      currentUserStats.new_last_active,
+      currentUserStats.client_id,
+    ]
+  );
 
-    created_at: toISO(row.goal_created_at),
-    days_interval: row.days_interval,
-    goal_reached_at: toISO(row.goal_reached_at),
+  const data = {
+    email: currentUserStats.email,
+
+    first_contribution_date: toISO(computed.first_contribution_date),
+    created_at: toISO(computed.goal_created_at),
+    days_interval: computed.days_interval,
+    goal_reached_at: toISO(computed.goal_reached_at),
 
-    last_active_date: toISO(row.last_active_date),
-    two_day_streak: Boolean(row.two_day_streak),
+    last_active_date: toISO(currentUserStats.new_last_active),
+    two_day_streak: Boolean(computed.two_day_streak),
   };
 
-  await sendRequest({
-    uri: API_URL + '/news/common-voice-goals/',
-    method: 'POST',
-    headers: {
-      'x-api-key': BASKET_API_KEY,
-    },
+  sendRequest({
+    ...basketConfig,
     form: data,
-  });
+  }).catch(err => console.error(err.message));
+
   console.log('basket', JSON.stringify(data, null, 2));
 }
+
+export async function sync(client_id: string) {
+  await computeGoals(client_id);
+  const currUserStats = await getCurrentStatus(client_id);
+
+  // If no newsletter prefs exist, they don't get email triggers
+  // If last update was within the past hour, do not send another sync
+  if (!currUserStats || currUserStats.hours_elapsed === 0) {
```

Comment:
I think this is fine but haven't tested yet – `hours_elapsed` will definitely be a number at this point, right? (as opposed to, eg, `"0"`)